### PR TITLE
Fix end to end testing for Forem forked repositories

### PR DESCRIPTION
**Pull request by nickytonline**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update
- [x] Testing infrastructure 

## Description

Fixes the issue with a Forem forked repository that does not have a Knapsack Pro API token that enables parallelization. If the API token is not found, `npx cypress run` is run instead.

## Related Tickets & Documents

#382, #11946

## QA Instructions, Screenshots, Recordings

Tests should run and pass.

### UI accessibility concerns?

N/A

## Added tests?

- [ ] Yes
- [x] No, and this is why: This PR is testing infrastructure
- [ ] I need help with writing tests

## Added to documentation?

- [ ] [Developer Docs](https://docs.forem.com) and/or
      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/)
- [ ] README
- [x] No documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

Ric Flair saying "Woo!"

### CODE REVIEW

File: .travis.yml
Comment (nickytonline) [inline .travis.yml:73]: I put this after the `FOREM_OWNER_SECRET` is set because it's required for the tests to run properly.

File: bin/e2e
Comment (nickytonline) [inline bin/e2e:1]: I modeled this script off of https://github.com/forem/forem/blob/15d975d1220c2aeacf02021014ba9c83d5aee474/bin/knapsack_pro_rspec. Thanks for pointing me to this @mstruve!

Comment (mstruve) [reply bin/e2e:1]: Below I noticed you are looking for the rspec token, even though technically either both will be there or they wont most of the time it would probably be a good idea to look for the Cypress token instead for the Cypress specs.

File: package.json
Comment (mstruve) [inline package.json:29]: nice!

**Review by mstruve — COMMENTED** [review]
Quick comment otherwise this looks good!

File: bin/e2e
Comment (nickytonline) [reply bin/e2e:1]: I'll make the change. Good catch @mstruve! I forgot there was a separate token for Cypress/Knapsack integration.

**Comment by rhymes** [comment]
What if we re-create this PR from an actual fork and test it before we merge? Because this is a branch and we won't know until we try it, right?

**Review by fdoxyz — APPROVED** [review]
Saw it working locally and the config steps in `.travis.yml` make sense to me!

I guess the trial by fire comes after we merge this and that's when we can make sure fork PRs still work as expected right?

**Comment by mstruve** [comment]
I like @rhymes's idea of testing this out quickly on a fork to make sure it works before we merge.

**Review by joshpuetz — APPROVED** [review]

**Comment by nickytonline** [comment]
> I like @rhymes's idea of testing this out quickly on a fork to make sure it works before we merge.

@rhymes @molly, I'll create a fork to test this.

**Pull request merged**
